### PR TITLE
[build-utils] ignore symlinked directories in glob

### PR DESCRIPTION
**Pull request by igorklopov**
Currently `glob` npm package has a flaw that allows following symlinked directories even if `follow` is not true. An example is pattern `**/node_modules/**`. Glob returns `node_modules/link-dir/node_modules/b.txt` among the results, tho `link-dir` is a symlink directory. In general this issue causes build cache extraction fail due to duplicate files (captured from both real dir and symlinked dir).

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [ ] This PR has a concise title and thorough description useful to a reviewer
- [ ] Issue from task tracker has a link to this PR

### CODE REVIEW

File: packages/build-utils/src/fs/glob.ts [hunk 1]
```diff
@@ -39,9 +39,30 @@ export default async function glob(
 
   options.symlinks = {};
   options.statCache = {};
+  options.cache = {};
   options.stat = true;
   options.dot = true;
 
+  await vanillaGlob(pattern, options); // populates the caches
```
Comment (AndyBitz) [inline packages/build-utils/src/fs/glob.ts:46]: Isn't it pretty inefficient if we call `glob` twice with the same pattern?

Comment (igorklopov) [reply packages/build-utils/src/fs/glob.ts:46]: I don't think so because the second run reuses the caches almost entirely

Comment (igorklopov) [reply packages/build-utils/src/fs/glob.ts:46]: Actually double-checking...

Comment (igorklopov) [reply packages/build-utils/src/fs/glob.ts:46]: Unfortunately glob still makes lstat calls even with full cache, so the second run is not in-memory only. Not enough "score" to continue here. The solution is going to be too hacky here. Will implement in another place of the infra.

**Pull request closed without merging**